filestream/filestream.go

```diff
@@ -0,0 +1,106 @@
+package filestream
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type FileStream struct {
+	file_name string
+	reader    *os.File
+	watcher   *fsnotify.Watcher
+
+	// Used to abort streaming when a client disconnects, etc
+	ctx context.Context
+}
+
+//-------------------------------------------------------------------------------------------------
+func New(ctx context.Context, file_name string) (*FileStream, error) {
+	file, err := os.Open(file_name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file '%s': %w", file_name, err)
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("failed to start a watcher: %w", err)
+	}
+
+	err = watcher.Add(file_name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add file '%s' to the watcher: %w", file_name, err)
+	}
+
+	return &FileStream{
+		file_name: file_name,
+		reader:    file,
+		watcher:   watcher,
+		ctx:       ctx,
+	}, nil
+}
+
+//-------------------------------------------------------------------------------------------------
+// Closes the fsnotify watcher and the underlying file stream
+func (s *FileStream) Close() error {
+	err := s.watcher.Close()
+	if err == nil {
+		err = s.reader.Close()
+	}
+	return err
+}
+
+//-------------------------------------------------------------------------------------------------
+func (s *FileStream) WaitForChanges() (changed bool) {
```

Comment:
I'm not sure you want to expose this publicly. Doing so could do two things:
1) allow a caller to spawn an endless number of goroutines
2) cause `Read` to potentially miss events from `fsnotify`

```suggestion
func (s *FileStream) waitForChanges() (changed bool) {
```